File: CodingSample/DotNet/Enterprise/Networking/TcpServerTest/DemoApp1/Models/ItemInfo.cs
namespace DemoApp.Models;

public readonly record struct ItemInfo(double Cost, int Stock)
{
    public static ItemInfo Parse(string message)
    {
        string[] segments = message.Split('&');
        double cost = double.Parse(segments[0][5..]);
        int stock = int.Parse(segments[1][6..]);
        return new ItemInfo(cost, stock);
    }
}

File: CodingSample/Java/Advanced/Networking/TCPClientTest/demo2/src/main/java/app/models/ItemInfo.java
package app.models;

public record ItemInfo(double cost, int stock) {
    
    public static ItemInfo parse(String message) {
        String[] parts = message.split("&");
        double c = Double.parseDouble(parts[0].substring(5));
        int s = Integer.parseInt(parts[1].substring(6));
        return new ItemInfo(c, s);
    }
}
